webapp/src/components/modals/create_issue/create_issue.tsx

```diff
@@ -0,0 +1,294 @@
+import {Post} from 'mattermost-redux/types/posts';
+
+import {getErrorMessage} from '../../../utils/user_utils';
+import GitlabLabelSelector from '../../gitlab_label_selector';
+import GitlabAssigneeSelector from '../../gitlab_assignee_selector';
+import GitlabMilestoneSelector from '../../gitlab_milestone_selector';
+import GitlabProjectSelector from '../../gitlab_project_selector';
+import Validator from '../../validator';
+import FormButton from '../../form_button';
+import Input from '../../input';
+
+const MAX_TITLE_LENGTH = 256;
+
+interface PropTypes {
+    post: Post | null,
+    title: string,
+    channelId: string,
+    theme: Theme,
+    visible: boolean,
+    actions: {
+        close: () => {
+            type: string;
+        };
+        create: (payload: IssueBody) => Promise<{
+            error?: ErrorType;
+        }>;
+    }
+};
+
+interface StateTypes {
+    submitting: boolean;
+    error: string | null;
+    project: ProjectSelection | null;
+    issueTitle: string;
+    issueDescription: string;
+    labels: SelectionType[];
+    assignees: SelectionType[];
+    milestone: SelectionType | null;
+    showErrors: boolean;
+    issueTitleValid: boolean;
+}
+
+const initialState = {
+    submitting: false,
+    error: null,
+    project: null,
+    issueTitle: '',
+    issueDescription: '',
+    labels: [],
+    assignees: [],
+    milestone: null,
+    showErrors: false,
+    issueTitleValid: true,
+};
+
+export default class CreateIssueModal extends PureComponent<PropTypes, StateTypes> {
+    validator: Validator
+
+    constructor(props: PropTypes) {
+        super(props);
+        this.state = initialState;
+        this.validator = new Validator();
+    }
+
+    componentDidUpdate(prevProps: PropTypes) {
+        if (this.props.post && !prevProps.post) {
+            this.setState({issueDescription: this.props.post.message});
+        } else if (this.props.channelId && (this.props.channelId !== prevProps.channelId || this.props.title !== prevProps.title)) {
+            const title = this.props.title.substring(0, MAX_TITLE_LENGTH);
+            this.setState({issueTitle: title});
+        }
+    }
+
+    // handle issue creation after form is populated
+    handleCreate = async (e: React.FormEvent<HTMLFormElement> | Event) => {
+        e.preventDefault();
+
+        if (!this.validator.validate() || !this.state.issueTitle) {
+            this.setState({
+                issueTitleValid: Boolean(this.state.issueTitle),
+                showErrors: true,
+            });
+            return;
+        }
+
+        const {post} = this.props;
+        const postId = post?.id ?? '';
+
+        const issue = {
+            title: this.state.issueTitle,
+            description: this.state.issueDescription,
+            project_id: this.state.project?.project_id,
+            labels: this.state.labels?.map((label) => label.value),
+            assignees: this.state.assignees?.map((assignee) => assignee.value),
```

Comment:
According to the Proptypes, labels and assignees cannot be null/undefined, then why use optional chaining here?